code-studio/src/js/levels/levelGroup.js

```diff
@@ -20,15 +20,57 @@ window.initLevelGroup = function (
 
   window.getResult = getResult;
 
+  function submitSublevelResults(completion, subLevelIdChanged) {
+    var levels = window.levelGroup.levels;
+    var sendReportCompleteCount = 0;
+    var subLevelCount = Object.keys(levels).length;
+    if (subLevelCount === 0) {
+      return completion();
+    }
+    for (var subLevelId in levels) {
+      if (typeof subLevelIdChanged !== 'undefined' && subLevelIdChanged !== parseInt(subLevelId)) {
+        // Only one sublevel changed and this is not the one, so skip the post and
+        // call the completion function immediately
+        completion();
```

Comment:
Confused: if we're looking through all the sublevels and we call completion() each time we encounter one that isn't the one we want, does this mean we're calling completion many times?